src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -347,6 +347,20 @@ def load_arguments(self, _):
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type,
                    deprecate_info=c.deprecate(target='--handle-extended-json-format/-j'))
         c.argument('no_prompt', arg_type=no_prompt)
+        c.argument('confirm_with_what_if', arg_type=deployment_what_if_confirmation_type, min_api="2019-10-01")
+        c.argument('what_if_result_format', options_list=['--what-if-result-format', '-r'],
+                   arg_type=deployment_what_if_result_format_type, min_api="2019-10-01")
+        c.argument('what_if_exclude_change_types', options_list=['--what-if-exclude-change-types', '-x'],
+                   arg_type=deployment_what_if_exclude_change_types_type,
+                   help="Space-separated list of resource change types to be excluded from What-If results. Applicable when --confirm-with-what-if is set.",
+                   min_api="2019-10-01")
+
+    with self.argument_context('deployment mg what-if') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
+        c.argument('no_prompt', arg_type=no_prompt)
+        c.argument('result_format', arg_type=deployment_what_if_result_format_type)
+        c.argument('no_pretty_print', arg_type=deployment_what_if_no_pretty_print_type)
+        c.argument('exclude_change_types', arg_type=deployment_what_if_exclude_change_types_type)
```

Comment:
Because these parameters are the same in different scopes, can they be configured centrally? 
such as:
```
  for scope in ['mg', 'tenant', 'sub', 'group']:
        with self.argument_context('deployment  {}  what-if'.format(scope)) as c:
                c.argument('deployment_name', arg_type=deployment_create_name_type)
                ......
```

Reply:
Although `deployment group` has a different set of parameters, `sub`, `mg` and `tenant` can be grouped together, and that is also the case for `create`, `validate`, `list`, etc. I also think that we should do some refactoring here, but we'd update other argument contexts, too, so I might leave that to a separate PR.

Reply:
Sounds good~